### DISCUSSION TITLE
Enhanced Statistics

### DISCUSSION THREAD

**Discussion by majora2007** [Kareadita/Kavita#3391] [Ideas]
### Idea Description

Kavita was always supposed to have more statistics. Let's use this space here to suggest different ideas. 

One idea I had that I thought would be amusing is seeing my stat of how many GBs of media I read this year or all time. 

Another idea in terms of a UX is using Githubs chart style for reading history instead of the line graph. 

### Idea Category

Feature Enhancement

### Duration of Using Kavita

Since inception

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by duplaja** [comment]
I really like the idea of the Github style heatmap for reading. I'm not sure how granular you want to get with things, but some other ideas might be:

* Week / Month / Year in review (top authors / genres / tags / library), possibly with a nice image of some sort to share.

* Average age of material read, over x time period: Are your users reading newly added materials? Ones that have been there for a while?

* Trends and streaks: Read x% more than last week. On an x day reading streak. 

* Maybe some kind of medal / badge system for certain benchmarks, for time read, number of items read, etc. (Possibly similar to Fitbit's distance ones, like Marathon: 26 Miles, London Underground: 250 Miles,Monarch Migration: 2,500 Miles, etc, but for time rather than distance). 

* More accurate customization of time spent reading. Let the use enter their own base words per minute reading speed, to override the default, if desired.

* Some way to see the intersection of time and reading choices. Ie: On weekends, you tend to read more romance novels. On weekday evenings, you read xianxia manga. During the month of December, you read more... Etc. And possibly expand out to time based recs (similar to Spotify's daylist concept, recommend things similar to what the user consumes at that time of day), as a longer term goal.

Just throwing out some ideas!

**Comment by scare376** [comment]
I'll toss the link to the comment i already made regarding some more stats from the epub reader.
https://github.com/Kareadita/Kavita/issues/1383#issuecomment-2316498636

**Comment by Sibeck11** [comment]
It might be worth reworking the read/unread functionality for this. Right now, it only tracks whether something is read or unread. This makes the "popular series" statistic that currently exists almost meaningless since it only adds +1 per account, not per actual times read. Implementing an actual counter could provide much more useful statistics.

If you currently finish a series, it always stays in the "continue" state even for the ones that aren't marked as "ongoing". I really like the idea of creating sort of "session data" for each read-through. It could be created when you first start a series and end when you reach the last page/chapter. This could give the ability to track not only how often you read a series, but could also track when you started/finished a series, how long it took you and all the other metadata you what to collect. With some extra background functions it could also tell you the read-time, average time per page and so on, which you could then compare to your global average from all the other books.

**Reply by majora2007** [reply]
Yeah this is something I've been thinking about. The implementation I chose works, but is limited in what we can do in terms of stats and re-read ability. Currently my thinking is to switch to a reading journal type system so I have a lot richer data on how many times a user has read each item and thus stats can be much richer. 

The problem is that is quite a refactor, so it needs careful attention (and I've been pushing out a lot of big refactors recently).

**Comment by yamanobe96** [comment]
I manage a library for my family.
I'd like to be able to check the reading history of other accounts. I'd like to be able to see which books in the library users are reading and add similar books to my library.
Of course, there are times when I listen to what people are reading and get reports of books they found interesting, but asking them every single time can be a burden for them.

**Comment by rmleonard** [comment]
How about stats per library/volume if I’m loading a library and it seems to be taking forever, it would be nice to know that it is on book 794/34276 ( maybe an arrow up or down to indicate count up or countdown). 

If I’m trying to use Mylar and it’s file structure as the back end, would there be any advantage to reading THAT database, datatable, config file etc, and then rather than completely reparsing each book based on its file name, directory name, etc. you simply do a table scan, verify that the book is there and copy out what you need.. and build stats as you go. It would seem (to me) that it would be faster and cut down on errors.

**Comment by duplaja** [comment]
One thing I've been kind of wanting recently was base library / smart filter level stats. (to clarify, this is not per user, but just flat "contained series stats")

Total read time / word counts for current library (or possibly even smart filter), in addition to number of series contained.

**Comment by Ark1369** [comment]
A cool feature for stats would be having an optional user fillable field called Read Year.
Once filled, it moves the "Read in X Year Stats"/"Word Count in X year" to the Year specified instead of year it was marked as read in.

This would help spread out stats naturally rather than 1 year having 100-1000s of series read when user joins Kavita.

**Comment by Ark1369** [comment]
Seperating this out, a Good stat for EPUBs to have would be something akin to what Moon Reader does - Character, Word Count per book. Hours spent reading, WPM etc
It also dives in deeper with Day Wise Breakdown per EPUB - Showing Date, Hours Read, WPM per day, but that is in realm of extended Information.
Example
![image](https://github.com/user-attachments/assets/9c9047a5-19ba-457b-9abe-735f474c288b)

**Comment by duplaja** [comment]
Something that came up today, in discussion. I'm not sure exactly how this is calculated for the graph, but it looks like the reading activity for epubs might just be dumping the total estimated reading length for the epub on the completion or last read date?

Context: I read a 5 million or so word epub for most of June and the start of July, finishing on July 7th, reading most days. The graph shows a 130 hour reading spike on the 7th, and almost nothing during the rest of June.  (0 for most days, when I usually read at least an hour or two for most)

<img width="1018" height="245" alt="image" src="https://github.com/user-attachments/assets/23623369-0e7d-4dd3-8053-3c9ad3c925d4" />
